File: packages/typescript/tests/system/executeJavascript.test.ts
import { ExecuteJavascriptTool } from "alumnium";
import { describe } from "vitest";
import { baseIt } from "./helpers.ts";

describe("Execute JavaScript", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);

      if (result.model.provider === "deepseek")
        skip("DeepSeek does not support vision yet");
      if (result.model.provider === "xai")
        skip("Requires a separate check agent");

      return result;
    };
  });

  it("scrolls the page", async ({ expect, setup }) => {
    const { al, $ } = await setup({
      extraTools: [ExecuteJavascriptTool],
    });
    await $.navigate("https://the-internet.herokuapp.com/large");

    await al.check("'Powered by Elemental Selenium' is not present", {
      vision: true,
      assert: expect.assert,
    });
    await al.do(
      "execute javascript 'window.scrollTo(0, document.body.scrollHeight)'",
    );
    await al.check("'Powered by Elemental Selenium' is present", {
      vision: true,
      assert: expect.assert,
    });
  });
});

File: packages/python/examples/pytest/execute_javascript_test.py
import pytest

from alumnium import Provider
from alumnium.tools import ExecuteJavascriptTool


def test_execute_javascript_to_scroll(al_factory, navigate):
    al = al_factory(extra_tools=[ExecuteJavascriptTool])
    model_provider = al.model.provider
    if model_provider == Provider.DEEPSEEK:
        pytest.xfail("No vision support yet")
    if model_provider == Provider.XAI:
        pytest.xfail(
            "Requires separate check agent as it prefers to follow "
            "retriever instructions (return `value` instead of `statement`)"
        )

    navigate("https://the-internet.herokuapp.com/large")
    al.check("'Powered by Elemental Selenium' is not present", vision=True)
    al.do("execute javascript 'window.scrollTo(0, document.body.scrollHeight)'")
    al.check("'Powered by Elemental Selenium' is present", vision=True)
